src/store/getters.ts

```diff
@@ -1,150 +1,95 @@
   },
 
-  reportState (state): string {
+  reportState (state: StateIF): string {
     switch (state.currentFilingStatus) {
       case FilingStatus.NEW: return ''
       case FilingStatus.DRAFT: return 'Draft'
     }
     return state.currentFilingStatus
   },
 
-  /** Returns date of last Change of Directors filing from list of past filings. */
-  // FUTURE: this will break if we only retrieve a page of filings
-  // FUTURE: the API should be giving us this date
-  lastCODFilingDate (state): string {
-    let lastCOD: string = null
-
-    for (let i = 0; i < state.filings.length; i++) {
-      let filing = state.filings[i].filing
-      // NB: these dates are UTC
-      let filingDate = filing.header.effectiveDate || filing.header.date
-      filingDate = filingDate.slice(0, 10)
-      if (filing.hasOwnProperty(FilingTypes.CHANGE_OF_DIRECTORS)) {
-        if (lastCOD === null || filingDate.split('-').join('') > lastCOD.split('-').join('')) {
-          lastCOD = filingDate
-        }
-      }
-    }
-    return lastCOD
-  },
-
-  /** Returns date of last Change of Address filing from list of past filings. */
-  // FUTURE: this will break if we only retrieve a page of filings
-  // FUTURE: the API should be giving us this date
-  lastCOAFilingDate (state): string {
-    let lastCOA: string = null
-
-    for (let i = 0; i < state.filings.length; i++) {
-      let filing = state.filings[i].filing
-      // NB: these dates are UTC
-      let filingDate = filing.header.effectiveDate || filing.header.date
-      filingDate = filingDate.slice(0, 10)
-      if (filing.hasOwnProperty(FilingTypes.CHANGE_OF_ADDRESS)) {
-        if (lastCOA === null || filingDate.split('-').join('') > lastCOA.split('-').join('')) {
-          lastCOA = filingDate
-        }
-      }
-    }
-    return lastCOA
-  },
-
-  /** Returns date of last filing (of any type) from list of past filings. */
-  // FUTURE: this will break if we only retrieve a page of filings
-  // FUTURE: the API should be giving us this date
-  lastFilingDate (state): string {
-    let lastFilingDate: string = null
-
-    for (let i = 0; i < state.filings.length; i++) {
-      let filing = state.filings[i].filing
-      // NB: these dates are UTC
-      let filingDate = filing.header.effectiveDate || filing.header.date
-      filingDate = filingDate.slice(0, 10)
-      if (lastFilingDate === null || filingDate.split('-').join('') > lastFilingDate.split('-').join('')) {
-        lastFilingDate = filingDate
-      }
-    }
-    return lastFilingDate
-  },
-
   /**
    * This is used to show Legal Obligations only for a new business
-   * that hasn't filed anything yet (except IA) and has no tasks.
+   * that has no tasks and hasn't filed anything yet (except their IA).
    **/
-  isBusinessWithNoMaintenanceFilings (state): boolean {
+  isBusinessWithNoMaintenanceFilings (state: StateIF): boolean {
     return (
+      // no todo items
+      state.tasks.length === 0 &&
```

Comment:
Because most languages -- including JS -- perform "short circuit evaluation", if the first condition of a && is false, it doesn't check the rest. (Similarly, if the first condition of a || is true, it doesn't check the rest.) Therefore, checking easy/fast stuff first makes the code just a little bit faster. It's an easy optimization :)